### PR TITLE
Update GeoJSON APIs to use Geo instead of Geometry

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR updates the GeoJSON implementation in `azure-core-experimental` in the following ways:

- Changed model types prefix from `Geometry` to `Geo`.
- Changed the package from `spatial` to `geojson`.
- Added `GeoArray` which acts as a generic representation of all coordinates in a `GeoObject` type.
- Removed direct access of `additionalProperties` to only allow property getting, renamed usage to `customProperties`.
- Added `GeoObjectType` enum which represents a `GeoObject`'s type.
- Added array-like getters to access `GeoObject` property coordinates along side the named getters.
- Added `GeoLinearRing` to represent a `GeoPolygon` type more closely to the RFC definition.

### CODE REVIEW

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoBoundingBox.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-package com.azure.core.experimental.spatial;
+package com.azure.core.experimental.geojson;
 
 import java.util.Objects;
 
 /**
  * Represents a geometric bounding box.
  */
-public final class GeometryBoundingBox {
+public final class GeoBoundingBox {
```
Comment (alzimmermsft) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoBoundingBox.java:11]: For the bounding box the RFC indicates that it must contain 2*N coordinates where N is equivalent to the number of coordinates in the geometry it is bounding. A 2D geometry requires the bounding box to have 4 coordinates, 3D requires 6.

https://tools.ietf.org/html/rfc7946#section-5

@pakrym, @KrzysztofCwalina, @JonathanGiles, @srnagar, would this be something we want to validate? I know our guidelines generally state to limit client side validation, is this more in the context of a service requires or does it apply more broadly to RFC guidelines/requirements as well?

Comment (pakrym) [reply sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoBoundingBox.java:11]: I think we should go with our guidance of not validating too much. We can let cosmos service decide whether it wants to accept these geometries.

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoArray.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.geojson;
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoArray.java:4]: Why do we need `json` in this package name? Can we just use `geo` or `geometry`? Not sure if we should be including the serialization format for these types.

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoBoundingBox.java (hunk 1 above)
Comment (srnagar) [reply sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoBoundingBox.java:11]: I agree. Let's not do too much validation here. Service can do this validation.

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoArray.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoArray.java:4]: As discussed offline `geojson` was chosen as geometries and spatial types extends outside of the scope of GeoJSON and that is what this implementation is targeting. cc: @pakrym 

For the serializers being in the package, that is done to scope feature work together in `azure-core-experimental`, once this functionality is merged into `azure-core` they'll exist in the serialization package.

**Pull request merged**